File: 290-word-pattern/290-word-pattern.js
/**
 * @param {string} pattern
 * @param {string} s
 * @return {boolean}
 */
var wordPattern = function(pattern, s) {
    const obj = {};
    const obj1 = {};
    s.split(' ').forEach((word, i) => {
        obj[word] = pattern[i];
    })
    pattern.split('').forEach((char, i) => {
        obj1[char] = s.split(' ')[i]
    })
    let newPattern = [];
    let newS = [];
    for (let i = 0; i < pattern.length; i++) {
        newPattern.push(obj[s.split(' ')[i]]);
        newS.push(obj1[pattern[i]]);
    }
    if (newPattern.join('') === pattern && newS.join(' ') === s) return true;
    return false;
};

File: 290-word-pattern/290-word-pattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        words_dict = {}
        chars_dict = {}
        words = s.split(' ')
        if len(words) != len(pattern): return False
        for char,word in zip(pattern, words):
            if char not in chars_dict:
                if word in words_dict:
                    return False
                else:
                    words_dict[word] = char
                    chars_dict[char] = word
            else:
                if chars_dict[char] != word: return False
        
        return True
                